### ISSUE TITLE
[Bug]: Share Image doesn't expose content-type

### ISSUE THREAD

**Issue by gjedeer** [mozilla-mobile/fenix#22403]
### Steps to reproduce

1. Go to https://www.meteo.pl/um/metco/mgram_pict.php?ntype=0u&row=411&col=207&lang=pl
2. Share Image (not URL)
3. The recieving application will not know that it's an image

### Expected behaviour

image/png MIME type is sent in the Intent and the app receiving knows it's an image. Telegram and Conversations both think that it's a PHP file.

### Actual behaviour

The intent being sent has `application/octet-stream` MIME type. The receiving application thinks it's a PHP file because it can only guess by extension.

Conversations developer iNPUTmice has identified it as:

```
cleanedUris=[content://org.mozilla.firefox.feature.downloads.fileprovider/mediaShareCache/1825338702.php], type=application/octet-stream
```

His full reply to the issue: https://github.com/iNPUTmice/Conversations/issues/4228#issuecomment-962579643

### Android version

Android 7

### Firefox release type

Firefox

### Firefox version

94.1.1

**Comment by Mugurell** [comment]
I think the share feature expects for an image url to end with an image extension.
If in this case the url ends with `.php` so the functionality gets a bit confused.
Saving from the same url works so it looks like we'd need something like https://github.com/mozilla-mobile/android-components/blob/6c0fe91ab83614265f533b819bf406fc046b2227/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/ext/DownloadState.kt#L32-L43 to fix this.

**Comment by Mugurell** [comment]
This should've been resolved in AC.

**Comment by Mugurell** [comment]
Should be available to test in the next Nightly.

**Comment by SoftVision-LorandJanos** [comment]
Verified as fixed on the latest Nightly 101.0a1 (2022-04-21).
Device used: Google Pixel 4 (Android 12).
Closing the ticket as fixed.

**Comment by gjedeer** [comment]
Thank you [Mugurell](https://github.com/Mugurell)!
